forms/OpenIDStep2Form.php

```diff
@@ -0,0 +1,321 @@
+                }
+            }
+        }
+
+        return parent::validate($callHooks);
+    }
+
+    /**
+     * @copydoc Form::execute()
+     */
+    function execute(...$functionArgs)
+    {
+        $register = is_string($this->getData('register'));
+        $connect = is_string($this->getData('connect'));
+        $oauthId = $this->getData('oauthId');
+        $selectedProvider = $this->getData('selectedProvider');
+        $result = false;
+
+        if (!empty($oauthId) && !empty($selectedProvider)) {
+            $oauthId = OpenIDHandler::encryptOrDecrypt($this->plugin, $this->contextId, 'decrypt', $oauthId);
+            // prevent saving one openid:ident to multiple accounts
+            $userIds = Repo::user()->getIdsBySetting('openid::'.$selectedProvider, $oauthId);
+            if ($userIds->isEmpty()) {
+                $userIds = Repo::user()->getIdsBySetting('openid::'.$selectedProvider, hash('sha256', $oauthId));
+            }
+            if ($userIds->isEmpty()) {
+                if ($register) {
+                    $user = $this->_registerUser();
+                    if (isset($user)) {
+                        if($selectedProvider == OpenIDPlugin::ORCID_PROVIDER)
+                        {
+                            $user->setOrcid($oauthId);
+                        }
+                        $user->setData('openid::'.$selectedProvider, $oauthId);
+                        
+                        Repo::user()->edit($user);
+                        $result = true;
+                    }
+                } elseif ($connect) {
+                    $payload = [
+                        'given_name' => $this->getData('givenName'), 
+                        'family_name' => $this->getData('familyName'), 
+                        'id' => $oauthId
+                    ];
+                    $username = $this->getData('usernameLogin');
+                    $password = $this->getData('passwordLogin');
+                    $user = Repo::user()->getByUsername($username, true);
+                    if (!isset($user)) {
+                        $user = Repo::user()->getByEmail($username, true);
+                    }
+                    if (isset($user) && Validation::verifyPassword($user->getUsername(), $password, $user->getPassword(), $rehash)) {
+                        $result = true;
+                    }
+                }
+                if ($result && isset($user)) {
+                    OpenIDHandler::updateUserDetails($this->plugin, isset($payload) ? $payload : null, $user, Application::get()->getRequest(), $selectedProvider, true);
+                    Validation::registerUserSession($user, $reason, true);
+                }
+            }
+        }
+        parent::execute(...$functionArgs);
+
+        return $result;
+    }
+
+
+    /**
+     * This function registers a new OJS User if no user exists with the given username, email or openid::{provider_name}!
```

Comment:
I'm guessing this plugin can be used in other applications without adaptation, right? If so, best not to refer to OJS.

Reply:
Removed `OJS` from here and elsewhere